offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/projects/ProjectDbConnector.groovy

```diff
@@ -0,0 +1,162 @@
+package life.qbic.portal.offermanager.dataresources.projects
+
+import groovy.util.logging.Log4j2
+import life.qbic.datamodel.dtos.business.Customer
+import life.qbic.datamodel.dtos.business.ProjectManager
+import life.qbic.datamodel.dtos.general.Person
+import life.qbic.portal.offermanager.dataresources.persons.PersonDbConnector
+
+import life.qbic.datamodel.dtos.projectmanagement.*
+
+import life.qbic.business.exceptions.DatabaseQueryException
+import life.qbic.portal.offermanager.dataresources.database.ConnectionProvider
+
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.Statement
+
+/**
+ * Provides operations on QBiC project data
+ *
+ * This class is responsible for transferring project data to the customer db
+ *
+ * @since: 1.0.0
+ *
+ */
+@Log4j2
+class ProjectDbConnector {
+
+  /**
+   * A connection to the customer database used to create queries.
+   */
+  private final ConnectionProvider connectionProvider
+  private final PersonDbConnector personDbConnector
+
+  /**
+   * Constructor for a ProjectDbConnector
+   * @param connectionProvider a connection provider to the customer db
+   * @param personDbConnector db connector used to connect projects to customer and manager
+   * 
+   */
+  ProjectDbConnector(ConnectionProvider connectionProvider, PersonDbConnector personDbConnector) {
+    this.connectionProvider = connectionProvider
+    this.personDbConnector = personDbConnector
+  }
+  
+  public Project addProjectAndConnectPersonsInUserDB(projectIdentifier, projectApplication) {
+    //collect infos needed for database
+    String projectTitle = projectApplication.getProjectTitle()
+    Customer customer = projectApplication.getCustomer()
+    ProjectManager projectManager = projectApplication.getProjectManager()
+    
+    //fetch needed person ids from database
+    int customerID = personDBConnector.getPersonId(customer)
+    int managerID = personDBConnector.getPersonId(projectManager)
+  
+    Connection connection = connectionProvider.connect()
+    connection.setAutoCommit(false)
+
+    connection.withCloseable {it ->
+      try {
+        int projectID = addProjectToDB(it, projectIdentifier, projectTitle)
+        addPersonToProject(it, projectID, managerID, "Manager")
+        addPersonToProject(it, projectID, customerID, "PI")
+      
+        it.commit()
+        
+      } catch (Exception e) {
+        log.error(e.message)
+        log.error(e.stackTrace.join("\n"))
+        it.rollback()
+        it.close() //is this needed?
+        throw new DatabaseQueryException("Could not add person and project data to user database.")
+      }
+    }
+  return new Project(projectIdentifier, projectTitle, projectApplication.getLinkedOffer())
+ }
+
+  private int isProjectInDB(String projectIdentifier) {
+    log.info("Looking for project " + projectIdentifier + " in the DB");
```

Comment:
maybe debug level, otherwise it will flood the log later

Reply:
done